src/watchmaker/utils/__init__.py

```diff
@@ -58,3 +59,37 @@ def urlopen_retry(uri):
         pass
 
     return urllib.request.urlopen(uri, **kwargs)
+
+
+def copytree(src, dst, force=False, ignore=False, **kwargs):
+    r"""
+    Copy OS directory trees from source to destination.
+
+    Args:
+        src: (:obj:`str`)
+            Source directory tree to be copied.
+            (*Default*: None)
+
+        dst: (:obj:`str`)
+            Destination where directory tree is to be copied.
+            (*Default*: None)
+
+        force: (:obj:`bool`)
+            Whether to delete destination prior to copy.
+            (*Default*: ``False``)
+
+        ignore: (:obj:`bool`)
+            Whether to ignore copy request if destination already exists.
+            (*Default*: ``False``)
+
+        Returns:
+            :obj:`bool`:
+                ``True`` if the copy was attempted. ``False`` if not.
+
+    """
+    if force and os.path.exists(dst):
+        shutil.rmtree(dst)
+    if not ignore or not os.path.exists(dst):
```

Comment:
This `ignore` seems a bit tortured. We're calling a function whose purpose is to copy a directory, but telling the function _not_ to copy the directory? This is the wrong place for that logic.